rust/smoketest/src/main.rs

```diff
@@ -0,0 +1,32 @@
+mod config;
+mod tests;
+
+use std::process::ExitCode;
+
+use crate::tests::*;
+use clap::Parser;
+use config::Args;
+use smoketest::new_client;
+
+#[tokio::main]
+async fn main() -> ExitCode{
```

Comment:
Generally I'm wondering why you wrote it as an program instead of reusing `rusts test ecosystem'? When using the test runner we don't have to setup the reporting by 'foot' via log messages and so on.